tree/dataframe/src/RLoopManager.cxx

```diff
@@ -301,6 +301,14 @@ DatasetLogInfo TreeDatasetLogInfo(const TTreeReader &r, unsigned int slot)
    return {std::move(what), static_cast<ULong64_t>(entryRange.first), end, slot};
 }
 
+static auto MakeDatasetColReadersKey(const std::string &colName, const std::type_info &ti)
+{
+   // We use a combination of column name and column type name as the key because in some cases we might end up
+   // with concrete readers that use different types for the same column, e.g. std::vector and RVec here:
+   //    df.Sum<vector<int>>("stdVectorBranch");
+   //    df.Sum<RVecI>("stdVectorBranch");
+   return colName + ':' + ti.name();
```

Comment:
colNames can never contain a `:`?

Reply:
Just mentioning `std::type_index`, fwiw.

Reply:
> colNames can never contain a :?

they can, but i'll still end up with different keys, e.g. `myColumn:int` and `myColumn::int`

> Just mentioning std::type_index, fwiw.

yeah but I need a key that contains both the column name _and_ the type name and I can't concatenate a `type_index` with the column name (and I didn't want to write a custom class with a custom hash function just for this..)

Reply:
oh wait, I _did_ write a custom class with a custom hash function just for this! and _then_ I decided it was overkill and stopped using it :sweat_smile: